core/services/pg/connection.go

```diff
@@ -94,6 +93,33 @@ func NewConnection(uri string, dialect dialects.DialectName, config ConnectionCo
 	return db, disallowReplica(db)
 }
 
+func setMaxConns(db *sqlx.DB, config ConnectionConfig) {
+	db.SetMaxOpenConns(config.MaxOpenConns())
+	db.SetMaxIdleConns(config.MaxIdleConns())
+
+	// HACK: In the case of mercury jobs, one conn is needed per job for good
+	// performance. Most nops will forget to increase the defaults to account
+	// for this so we detect it here instead.
+	//
+	// This problem will be solved by replacing mercury with parallel
+	// compositions (llo plugin).
+	//
+	// See: https://smartcontract-it.atlassian.net/browse/MERC-3654
+	var cnt int
+	if err := db.Get(&cnt, `SELECT COUNT(*) FROM ocr2_oracle_specs WHERE plugin_type = 'mercury'`); err != nil {
+		log.Printf("Error checking mercury jobs: %s", err.Error())
+		return
+	}
+	if cnt > config.MaxOpenConns() {
```

Comment:
I think this is a good solution until parallel comp, but but 1 job per connection might cause issues if we have 1000 jobs

Reply:
We can't run that many jobs on one node performantly then, would need to have multiple DONs